internal/pkg/otel/monitoring/monitoring.go

```diff
@@ -0,0 +1,134 @@
+// Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+// or more contributor license agreements. Licensed under the Elastic License 2.0;
+// you may not use this file except in compliance with the Elastic License 2.0.
+
+package monitoring
+
+import (
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"net/url"
+	"os"
+	"path/filepath"
+	"runtime"
+	"strings"
+
+	"github.com/gofrs/uuid/v5"
+	"github.com/gorilla/mux"
+
+	"github.com/elastic/elastic-agent-libs/api"
+	"github.com/elastic/elastic-agent-libs/logp"
+	"github.com/elastic/elastic-agent-libs/monitoring"
+	"github.com/elastic/elastic-agent-system-metrics/report"
+	"github.com/elastic/elastic-agent/internal/pkg/agent/application/paths"
+	"github.com/elastic/elastic-agent/pkg/utils"
+	"github.com/elastic/elastic-agent/version"
+)
+
+// EDOTComponentID is the component ID for the EDOT collector.
+const EDOTComponentID = "elastic-agent/collector"
+
+// EDOTMonitoringEndpoint returns the monitoring endpoint for the EDOT collector.
+func EDOTMonitoringEndpoint() string {
+	return utils.SocketURLWithFallback(EDOTComponentID, paths.TempDir())
+}
+
+// NewServer creates a new server exposing metrics and process information.
+func NewServer(log *logp.Logger, host string) (*api.Server, error) {
+	ephemeralID, err := generateEphemeralID()
+	if err != nil {
+		return nil, err
+	}
+
+	if err := report.SetupMetricsOptions(report.MetricOptions{
+		Name:           EDOTComponentID,
+		Version:        version.GetDefaultVersion(),
+		Logger:         log,
+		EphemeralID:    ephemeralID,
+		SystemMetrics:  monitoring.Default.GetOrCreateRegistry("system"),
+		ProcessMetrics: monitoring.Default.GetOrCreateRegistry("beat"),
+	}); err != nil {
+		return nil, fmt.Errorf("failed to setup metrics: %w", err)
+	}
+
+	r := mux.NewRouter()
+	r.HandleFunc("/stats", func(w http.ResponseWriter, r *http.Request) {
```

Comment:
Why can't we use the same stats handler as the agent's monitoring server?

Reply:
My initial effort was actually to utilise the same code as found [here](https://github.com/elastic/elastic-agent/blob/4cd10aa641a5e298dabed246f1f55b98fbcd6afa/internal/pkg/agent/application/monitoring/server.go#L30). However this code is not modular so we would get way more than just `/stats` but more importantly it doesn't allow me to control the socket/npipe path. Now instead of making the existing code modular and, thus, risking of changing something that might lead to issues with code that currently works, I chose to extract what was needed from there and invoke a monitoring server with just the `/stats` path exposed. Also keeping in mind that this endpoint might become part of EDOT diagnostics extension, this seemed like the less radical way to proceed. WDYT?

Reply:
Allright, that works.